contracts/BSCValidatorSet.sol

```diff
@@ -450,15 +459,49 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return isWorkingValidator(index);
   }
 
+  function distributeFinalityReward(address[] calldata valAddrs, uint256[] calldata weights) external onlyCoinbase onlyInit {
+    uint256 totalValue;
+    totalValue = (address(SYSTEM_REWARD_ADDR).balance * finalityRewardRatio) / 100;
+    totalValue = ISystemReward(SYSTEM_REWARD_ADDR).claimRewards(payable(address(this)), totalValue);
```

Comment:
Validator contract is not allowed to claim reward from system reward contract, you need come up a way to enable it.

Reply:
Fixed